oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -727,26 +604,22 @@ public void deleteAllMetadataRecords(String prefix) {
         try {
             Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
 
-            CloudBlobDirectory metaDir = getAzureContainer().getDirectoryReference(META_DIR_NAME);
             int total = 0;
-            for (ListBlobItem item : metaDir.listBlobs(prefix)) {
-                if (item instanceof CloudBlob) {
-                    if (((CloudBlob)item).deleteIfExists()) {
-                        total++;
-                    }
+
+            for (BlobItem blobItem : getAzureContainer().listBlobsByHierarchy(META_DIR_NAME)) {
```

Comment:
same as above:
`listBlobsByHierarchy` is listing only the hierarchy, so `META/`.
this has to be changed with
```
ListBlobsOptions listBlobsOptions = new ListBlobsOptions();
listBlobsOptions.setPrefix(META_DIR_NAME);
getAzureContainer().listBlobs(listBlobsOptions, null);
```